### PR TITLE
Docs: Update Callout documentation, add best practices

### PR DESCRIPTION
**Pull request by ayeshakmaz**
Update Callout docs to new docs

### CODE REVIEW

**Comment by ashleyseto** [comment]
I've updated the paper doc with visual examples for the Do and Don't section. I think we need them otherwise my eyes just skip that section and go straight to accessibility.

For another PR, I'd like to come up with some other styling for the props in text. They still kind of blend in.

**Comment by ashleyseto** [comment]
I think there is some extra space happening in some of the examples.
space

---

@anniemochi I think we need to adjust the space between examples. Probably add more since it looks like they're blending together.

File: docs/src/Callout.doc.js [hunk 1]
```diff
@@ -22,168 +47,354 @@ card(
...
         required: true,
         defaultValue: null,
-        description: [
-          'Text to render inside the callout to convey detailed information to the user. The message text has a fixed size.',
-        ],
-        href: '',
+        description: `
+          Main content of the Callout. Content should be localized.
+
+          See [Best Practices](#Best-practices) for more info`,
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:53]: period at the end?

File: docs/src/Callout.doc.js [hunk 2]
```diff
@@ -22,168 +47,354 @@ card(
         type: 'string',
         required: true,
         defaultValue: null,
-        description: [
-          'Text to render inside the callout to convey detailed information to the user. The message text has a fixed size.',
-        ],
-        href: '',
+        description: `
+          Main content of the Callout. Content should be localized.
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:51]: no article

File: docs/src/Callout.doc.js [hunk 3]
```diff
@@ -22,168 +47,354 @@ card(
...
-        required: false,
         defaultValue: null,
-        description: [
-          'Callback fired when the dismiss button is clicked (pressed and released) with a mouse or keyboard.',
-          'Supply a short, descriptive label for screen-readers to provide sufficient context about the dismiss button action. IconButtons do not render text for screen readers to read requiring an accessibility label.',
-          'Accessibility: `accessibilityLabel` populates aria-label.',
-        ],
-        href: '',
+        description: `
+          When specified, adds a dismiss button to the Callout. See the [Dismissible variant](#Dismissible) for more info.
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:60]: no article

File: docs/src/Callout.doc.js [hunk 4]
```diff
@@ -22,168 +47,354 @@ card(
...
         required: true,
         defaultValue: null,
-        description: [
-          'Supply a short, descriptive label for screen-readers to provide sufficient context about the callout intent.',
-          'The callout component displays `error`, `info`, or `warning` icons which visually indicate the intent of the component. `iconAccessibilityLabel` allows screen readers to communicate that intent.',
-          'Accessibility: It populates aria-label.',
-        ],
-        href: '',
+        description: `
+          Label to describe the icon’s purpose
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:70]: This format is funky. Should be two sentences inline, with periods at their ends?

File: docs/src/Callout.doc.js [hunk 5]
```diff
@@ -22,168 +47,354 @@ card(
...
-          '- label: Text to render inside the button to convey the function and purpose of the button. The button text has a fixed size.',
-          '- accessibilityLabel: Supply a short, descriptive label for screen-readers to replace button texts that do not provide sufficient context about the button component behavior. Texts like `Click Here,` `Follow,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the button text.',
-          '- onClick: Callback fired when the button component is clicked (pressed and released) with a mouse or keyboard.',
-          `- onNavigationOptions: onNavigationOptions works in conjunction with a Provider. Pass custom props to onNavigation. See Provider for examples. onNavigation's type is flexible. Each key's value is a React.Node or an event handler function. Optional with href.`,
-          'Accessibility: `accessibilityLabel` populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the button `text`.',
-        ],
+        description: `
+          Main action for people to take on the Callout
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:81]: ditto here on funky layout, and this first sentence needs a period (and no article)

File: docs/src/Callout.doc.js [hunk 6]
```diff
@@ -22,168 +47,354 @@ card(
...
-          '- label: Text to render inside the button to convey the function and purpose of the button. The button text has a fixed size.',
-          '- accessibilityLabel: Supply a short, descriptive label for screen-readers to replace button texts that do not provide sufficient context about the button component behavior. Texts like `Click Here,` `Follow,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the button text.',
-          '- onClick: Callback fired when the button component is clicked (pressed and released) with a mouse or keyboard.',
-          `- onNavigationOptions: onNavigationOptions works in conjunction with a Provider. Pass custom props to onNavigation. See Provider for examples. onNavigation's type is flexible. Each key's value is a React.Node or an event handler function. Optional with href.`,
-          'Accessibility: `accessibilityLabel` populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the button `text`.',
-        ],
+        description: `
+          Secondary action for people to take on the Callout
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:95]: ditto here

File: docs/src/Callout.doc.js [hunk 7]
```diff
@@ -22,168 +47,354 @@ card(
...
       {
         name: 'title',
         type: 'string',
-        required: false,
         defaultValue: null,
-        description: [
-          'Heading to render inside the callout above the message to convey the callout topic to the user. The message text has a fixed size.',
-        ],
-        href: '',
+        description: `Brief title summarizing the Callout. Content should be localized.`,
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:116]: no article

File: docs/src/Callout.doc.js [hunk 8]
```diff
@@ -22,168 +47,354 @@ card(
...
         defaultValue: null,
-        description: [
-          'Select a type of callout:',
-          '-`error`: To provide error messaging to the user.',
-          '-`info`: To provide general information to the user.',
-          '-`warning`: To provide a warn message to the user.',
-        ],
-        href: '',
+        description: `
+        The type of Callout. See the [Variants](#Variants) to learn more
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:109]: period at the end

File: docs/src/Callout.doc.js [hunk 9]
```diff
@@ -22,168 +47,354 @@ card(
...
+            iconAccessibilityLabel="Error"
+            message="Your tag has errors, so information may be outdated. Fix your tag for the most accurate metrics."
+            primaryAction={{href: "https://pinterest.com", label:"Fix tag"}}
+
+          />
+        `}
+      />
+      <MainSection.Card
+        cardSize="lg"
+        type="do"
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:142]: I know it's different given the wider content, but it seems odd to me to not alternate Do's and Don't's

File: docs/src/Callout.doc.js [hunk 10]
```diff
@@ -22,168 +47,354 @@ card(
...
+            </Box>
+            <Box height={150}/>
+          </Box>
+        `}
+      />
+      <MainSection.Card
+        cardSize="lg"
+        type="don't"
+        description={`
+        Use Callouts for marketing new products or features. Use an [Upsell](/Upsell) instead.
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:179]: no article

File: docs/src/Callout.doc.js [hunk 11]
```diff
@@ -22,168 +47,354 @@ card(
...
+    </MainSection.Subsection>
+  </MainSection>,
+);
+
+card(
+  <MainSection name="Accessibility">
+    <MainSection.Subsection
+      title="Labels"
+      description={`
+      The \`iconAccessibilityLabel\`, \`dismissButton\`, \`primaryAction\`, and \`secondaryAction\` each require a short, descriptive label for screen readers. These labels should communicate intent or visually describe what any icons might mean, such as “error”, “info” or “warning”. They should also be localized.
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:250]: no article at the beginning of the first sentence

File: docs/src/Callout.doc.js [hunk 12]
```diff
@@ -22,168 +47,354 @@ card(
...
+
+      In the case of [Buttons](/Button), alternative text should be provided to replace vague text like "Visit" or "Learn more" with more descriptive information, like ‘Learn more about our work from home resources’.
+
+      For [IconButtons](/IconButtons), like the \`dismissButton\`, the label provided should indicate the intent, like “Dismiss this banner”.
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:254]: This wording makes it sound like we're providing `dismissButton` as an example of IconButton (which it is, but that's irrelevant). I think what we're going for here is more "similar to `dismissButton`…"

File: docs/src/Callout.doc.js [hunk 13]
```diff
@@ -22,168 +47,354 @@ card(
...
+  </MainSection>,
 );
 
 card(
-  <Example
-    name="Warning Example"
-    defaultCode={`
+  <MainSection
+    name="Localization"
+    description={`Remember to localize all Link or Button labels, as well as the \`title\` and \`message\`.`}
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:281]: no articles on props

File: docs/src/Callout.doc.js [hunk 14]
```diff
@@ -22,168 +47,354 @@ card(
...
+        cardSize="lg"
+        defaultCode={`
 <Callout
-  type="warning"
-  iconAccessibilityLabel="Warning icon"
-  message="This feature will be removed in two weeks."
-  primaryAction={{href: "https://pinterest.com", label:"Learn more"}}
+  type="info"
+  iconAccessibilityLabel="Info"
+  title="Ihr Geschäftskonto wurde erstellt!"
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:290]: German is a nice touch! 👌

File: docs/src/Callout.doc.js [hunk 15]
```diff
@@ -267,7 +451,53 @@ function Example(props) {
...
   );
 }
-`}
+        `}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Dismissible"
+      description={`
+        A \`dismissButton\` can be used when a Callout does not indicate a persistent state. This will most commonly be used in info Callouts.
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:460]: no article at the beginning of the first sentence

File: docs/src/Callout.doc.js [hunk 16]
```diff
@@ -267,7 +451,53 @@ function Example(props) {
...
+  </MainSection>,
+);
+
+card(
+  <MainSection
+    name="Related"
+    description={`
+      [Upsell](/upsell)
+
+      If marketing new products or features, or encouraging upgrades, use an Upsell instead.
```
Comment (dangerismycat) [inline docs/src/Callout.doc.js:495]: no article

**Review by dangerismycat — COMMENTED** [review]
Aside from the question about alternating Do's and Don't's, just some copy nits. This is looking awesome!

**Comment by AlbertCarreras** [comment]
Out of scope for this PR, but I feel the top/bottom spaces in the main example look unbalanced to me.
image

File: docs/src/Callout.doc.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/src/Callout.doc.js:51]: I removed all articles from Tooltip

File: docs/src/Callout.doc.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/src/Callout.doc.js:81]: Main action for _users_? to take on the Callout

File: docs/src/Callout.doc.js [hunk 17]
```diff
@@ -22,168 +47,354 @@ card(
...
+    accessibilityLabel: 'Dismiss this banner',
+    onDismiss: ()=>{},
+  }}
+/>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Warning"
+      description="Warning Callouts communicate cautionary messages to users. Action should not be required. The Callout should provide clear guidance on how to correct an issue and/or learn more about it."
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:331]: shouldn't  as per writing guidelines

File: docs/src/Callout.doc.js [hunk 18]
```diff
@@ -22,168 +47,354 @@ card(
...
+    accessibilityLabel: 'Dismiss warning',
+    onDismiss: ()=>{},
+  }}
+/>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Error"
+      description="Error Callouts inform users of problems that require immediate action to correct. Further actions on the page might be blocked if users do not correct the problems. The Callout should also provide clear guidance on how to correct the issue and/or learn more about it."
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:352]: don't  as per writing guidelines

File: docs/src/Callout.doc.js [hunk 19]
```diff
@@ -22,168 +47,354 @@ card(
...
-  `}
-  />,
-);
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Actions"
+      description={`
+        Callouts can have either one primary action, or a secondary action and a primary action. These actions can be [Links](/Link), by specifying the \`href\` property, or [Buttons](/Buttons), when no \`href\` is supplied.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:371]: not strong about this, but 

        `Callouts can have either one primary action, or a _primary_ action and a _secondary_ action.`

File: docs/src/Callout.doc.js [hunk 20]
```diff
@@ -22,168 +47,354 @@ card(
...
+        iconAccessibilityLabel="Info"
+        title="Your business account was created!"
+        message="Apply to the Verified Merchant Program"
         primaryAction={{
-          label:"Edit board", role: "button",
+          label:"Apply now",
           onClick: () => { setShowModal(!showModal) }
         }}
+        secondaryAction={{
+          href: "https://help.pinterest.com/en/business/article/verified-merchant-program",
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:394]: target='blank'?

File: docs/src/Callout.doc.js [hunk 21]
```diff
@@ -267,7 +451,53 @@ function Example(props) {
...
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Dismissible"
+      description={`
+        A \`dismissButton\` can be used when a Callout does not indicate a persistent state. This will most commonly be used in info Callouts.
+
+        Do not use dismiss buttons in the following cases:
+
+        - There is a persistent account or page status that the user must address.
+        - The user must access this information again in order to perform a task.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:465]: not strong about it but could we make a little bit more explicit what  `this information` means?

File: docs/src/Callout.doc.js [hunk 22]
```diff
@@ -267,7 +451,53 @@ function Example(props) {
...
+    name="Related"
+    description={`
+      [Upsell](/upsell)
+
+      If marketing new products or features, or encouraging upgrades, use an Upsell instead.
+
+      [Toast](/toast)
+
+      Toast provides feedback on a user interaction, like a confirmation that appears when a Pin has been saved. Unlike Upsell and Callout, Toasts don’t contain actions. They’re also less persistent, and typically disappear after a certain duration.
+    `}
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:500]: How do we feel about adding `ActivationCard` here? I feel ActivationCard could be more relevant than Toast.

File: docs/src/Callout.doc.js (hunk 5 above)
Comment (ayeshakmaz) [reply docs/src/Callout.doc.js:81]: Number 11 [here](https://paper.dropbox.com/doc/How-to-Convert-docs-to-new-design--BFWjPEnYkZkNBUU7ZimX79JBAg-6jXGUn10D7Ug258gdUmkp#:uid=315708423677403158321907&h2=%E2%86%92-Be-sure-to-refer-to-Pinteres) suggests it should be people. @anniemochi what do you think?

File: docs/src/Callout.doc.js (hunk 9 above)
Comment (ayeshakmaz) [reply docs/src/Callout.doc.js:142]: Chatted with @anniemochi about this and I think the guideline is to stack Dos then Don'ts if they're large. Annie what do you think?

File: docs/src/Callout.doc.js (hunk 14 above)
Comment (AlbertCarreras) [reply docs/src/Callout.doc.js:282]: Should Verified Merchant Program be transalted too?

File: docs/src/Callout.doc.js [hunk 23]
```diff
@@ -22,168 +47,346 @@ card(
         type: 'string',
         required: true,
         defaultValue: null,
-        description: [
-          'Text to render inside the callout to convey detailed information to the user. The message text has a fixed size.',
-        ],
-        href: '',
+        description: `
+          Main content of Callout. Content should be localized.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:51]: Should we standarize it to `See [localization ](#localization) to learn more.`

File: docs/src/Callout.doc.js [hunk 24]
```diff
@@ -22,168 +47,346 @@ card(
...
         defaultValue: null,
-        description: [
-          'Select a type of callout:',
-          '-`error`: To provide error messaging to the user.',
-          '-`info`: To provide general information to the user.',
-          '-`warning`: To provide a warn message to the user.',
-        ],
-        href: '',
+        description: `
+        The type of Callout. See the [Variants](#Variants) to learn more.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:103]: the word `type` might be redundant. `Callout categories.` I was thinking that this is like traffic signs, each style conveys different categories, different information messages.

File: docs/src/Callout.doc.js (hunk 9 above)
Comment (dangerismycat) [reply docs/src/Callout.doc.js:136]: Ah ok, that's fine with me if that's our standard. 👍

File: docs/src/Callout.doc.js [hunk 25]
```diff
@@ -22,168 +47,346 @@ card(
...
       {
         name: 'title',
         type: 'string',
-        required: false,
         defaultValue: null,
-        description: [
-          'Heading to render inside the callout above the message to convey the callout topic to the user. The message text has a fixed size.',
-        ],
-        href: '',
+        description: `Brief title summarizing Callout. Content should be localized.`,
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:110]: ditto re  Localization

File: docs/src/Callout.doc.js [hunk 26]
```diff
@@ -22,168 +47,346 @@ card(
...
-        required: false,
         defaultValue: null,
-        description: [
-          'Callback fired when the dismiss button is clicked (pressed and released) with a mouse or keyboard.',
-          'Supply a short, descriptive label for screen-readers to provide sufficient context about the dismiss button action. IconButtons do not render text for screen readers to read requiring an accessibility label.',
-          'Accessibility: `accessibilityLabel` populates aria-label.',
-        ],
-        href: '',
+        description: `
+          When specified, adds a dismiss button to Callout. See the [Dismissible variant](#Dismissible) for more info.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:60]: I wonder if `When specified, ` is redundant as `Adds` conveys the same optional/additional nature.

File: docs/src/Callout.doc.js [hunk 27]
```diff
@@ -22,168 +47,346 @@ card(
...
-          '- accessibilityLabel: Supply a short, descriptive label for screen-readers to replace button texts that do not provide sufficient context about the button component behavior. Texts like `Click Here,` `Follow,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the button text.',
-          '- onClick: Callback fired when the button component is clicked (pressed and released) with a mouse or keyboard.',
-          `- onNavigationOptions: onNavigationOptions works in conjunction with a Provider. Pass custom props to onNavigation. See Provider for examples. onNavigation's type is flexible. Each key's value is a React.Node or an event handler function. Optional with href.`,
-          'Accessibility: `accessibilityLabel` populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the button `text`.',
-        ],
+        description: `
+          Main action for people to take on Callout. If \`href\` is supplied, the action will serve as a link. If no \`href\` is supplied, the action will be a button.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:79]: s/people/users ?

File: docs/src/Callout.doc.js [hunk 28]
```diff
@@ -22,168 +47,346 @@ card(
...
-          '- accessibilityLabel: Supply a short, descriptive label for screen-readers to replace button texts that do not provide sufficient context about the button component behavior. Texts like `Click Here,` `Follow,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the button text.',
-          '- onClick: Callback fired when the button component is clicked (pressed and released) with a mouse or keyboard.',
-          `- onNavigationOptions: onNavigationOptions works in conjunction with a Provider. Pass custom props to onNavigation. See Provider for examples. onNavigation's type is flexible. Each key's value is a React.Node or an event handler function. Optional with href.`,
-          'Accessibility: `accessibilityLabel` populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the button `text`.',
-        ],
+        description: `
+          Secondary action for people to take on Callout. If \`href\` is supplied, the action will serve as a link. If no \`href\` is supplied, the action will be a button.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:91]: s/people/users ?

File: docs/src/Callout.doc.js [hunk 29]
```diff
@@ -22,168 +47,346 @@ card(
...
+            primaryAction={{href: "https://pinterest.com", label:"Fix tag"}}
+
+          />
+        `}
+      />
+      <MainSection.Card
+        cardSize="lg"
+        type="do"
+        description={`
+        Place the Callout at the top of the page under the primary navigation or page header when possible.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:138]: Ryan didnt get this `the`

File: docs/src/Callout.doc.js [hunk 30]
```diff
@@ -22,168 +47,346 @@ card(
...
+            }}
+          />
+        `}
+      />
+
+      <MainSection.Card
+        cardSize="lg"
+        type="don't"
+        description={`
+        Stack Callouts. In the case that banners must be stacked, Callouts will usually come before Upsells.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:193]: I wonder if `will usually` is confusing.  What about `Callouts come before Upsells`

File: docs/src/Callout.doc.js (hunk 29 above)
Comment (dangerismycat) [reply docs/src/Callout.doc.js:138]: 😅

File: docs/src/Callout.doc.js [hunk 31]
```diff
@@ -22,168 +47,346 @@ card(
...
+    </MainSection.Subsection>
+  </MainSection>,
+);
+
+card(
+  <MainSection name="Accessibility">
+    <MainSection.Subsection
+      title="Labels"
+      description={`
+      \`iconAccessibilityLabel\`, \`dismissButton\`, \`primaryAction\`, and \`secondaryAction\` each require a short, descriptive label for screen readers. These labels should communicate intent or visually describe what any icons might mean, such as “error”, “info” or “warning”. They should also be localized.
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:242]: visually describe? what does visually describe mean?

Comment (AlbertCarreras) [reply docs/src/Callout.doc.js:242]: Are `These labels` referring to `iconAccessibilityLabel`. Maybe: `The `error`, `info` or `warning` Icon labels should communicate intent or describe what any icons might mean, such as X X X

Not strong about it, just a thought.

File: docs/src/Callout.doc.js [hunk 32]
```diff
@@ -22,168 +47,346 @@ card(
...
+  </MainSection>,
 );
 
 card(
-  <Example
-    name="Warning Example"
-    defaultCode={`
+  <MainSection
+    name="Localization"
+    description={`Remember to localize all Link or Button labels, as well as \`title\` and \`message\`.`}
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:273]: Remember to localize all **l**ink or **b**utton labels ? we arent referring to the components itself, no?

File: docs/src/Callout.doc.js [hunk 33]
```diff
@@ -22,168 +47,346 @@ card(
...
+      title="Warning"
+      description="Warning Callouts communicate cautionary messages to users. Action shouldn't be required. The Callout should provide clear guidance on how to correct an issue and/or learn more about it."
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+<Callout
+  type="warning"
+  iconAccessibilityLabel="Warning"
+  title="Unused ActAlike audiences are going away"
```
Comment (AlbertCarreras) [inline docs/src/Callout.doc.js:331]: ActAlike can we replace it with something simpler? it's hard to comprehend

**Review by dangerismycat — APPROVED** [review]
:shipit:

File: docs/src/Callout.doc.js (hunk 23 above)
Comment (ayeshakmaz) [reply docs/src/Callout.doc.js]: I ended up linking localized to the localization section

**Pull request merged**